### PR TITLE
Remove Mode and customizable base url 

### PR DESCRIPTION
**Pull request by M1r1k**

### CODE REVIEW

File: tests/SmartlingApiTest.php [hunk 1]
```diff
@@ -163,14 +162,14 @@ public function constructorDataProvider()
...
-            ['uri', 'api-key', 'product-id', $mockedClient, SmartlingApi::SANDBOX_MODE, SmartlingApi::SANDBOX_URL],
-            ['uri', 'api-key', 'product-id', $mockedClient, SmartlingApi::PRODUCTION_MODE, 'uri'],
-            ['uri', 'api-key', 'product-id', $mockedClient, null, SmartlingApi::SANDBOX_URL],
-            ['uri', 'api-key', 'product-id', $mockedClient, 'unknown', SmartlingApi::SANDBOX_URL],
-            ['', 'api-key', 'product-id', $mockedClient, 'unknown', SmartlingApi::SANDBOX_URL],
-            ['', 'api-key', 'product-id', $mockedClient, SmartlingApi::PRODUCTION_MODE, SmartlingApi::PRODUCTION_URL],
-            ['', 'api-key', 'product-id', $mockedClient, SmartlingApi::SANDBOX_MODE, SmartlingApi::SANDBOX_URL],
-            ['', 'api-key', 'product-id', $mockedClient, null, SmartlingApi::SANDBOX_URL],
+            ['api-key', 'product-id', $mockedClient],
+//            ['uri', 'api-key', 'product-id', $mockedClient, SmartlingApi::PRODUCTION_MODE, 'uri'],
```
Comment (andypost) [inline tests/SmartlingApiTest.php]: looks this data provider is not needed anymore

File: src/SmartlingApi.php [hunk 2]
```diff
@@ -122,7 +109,9 @@ protected function sendRequest($uri, $requestData, $method)
             }
         }
 
-        $guzzle_response = $this->httpClient->request($method, $this->baseUrl . '/' . $uri, $options);
+        $uri = ltrim($uri, "/");
```
Comment (andypost) [inline src/SmartlingApi.php]: this needs code comment

File: src/SmartlingApi.php [hunk 3]
```diff
@@ -122,7 +109,9 @@ protected function sendRequest($uri, $requestData, $method)
...
+
+        $guzzle_response = $this->httpClient->request($method, self::SERVICE_URL . $uri, $options);
```
Comment (andypost) [inline src/SmartlingApi.php]: This kills ability to configure URL or use different server

File: src/SmartlingApi.php [hunk 4]
```diff
@@ -7,10 +7,7 @@
 class SmartlingApi
 {
 
-    const SANDBOX_MODE = 'SANDBOX';
-    const PRODUCTION_MODE = 'PRODUCTION';
-    const SANDBOX_URL = 'https://sandbox-api.smartling.com/v1/';
-    const PRODUCTION_URL = 'https://api.smartling.com/v1/';
+    const SERVICE_URL = 'https://api.smartling.com/v1/';
```
Comment (andypost) [inline src/SmartlingApi.php]: This is defaults, not hardcoded, so should be injected with constructor

File: src/SmartlingApi.php [hunk 5]
```diff
@@ -43,27 +40,17 @@ class SmartlingApi
     /**
      * Creates SmartlingApi instance.
      *
-     * @param string $baseUrl
```
Comment (andypost) [inline src/SmartlingApi.php]: Removing this does not allow to change server URL anymore

File: src/SmartlingApi.php [hunk 6]
```diff
@@ -43,27 +40,17 @@ class SmartlingApi
...
      *   Api Key string.
      * @param string $projectId
      *   Project ID string.
      * @param \GuzzleHttp\ClientInterface $http_client
      *   Instance of Guzzle http client.
-     * @param string $mode
-     *   Production or Sandbox mode.
      */
-    public function __construct($baseUrl, $apiKey, $projectId, ClientInterface $http_client, $mode = self::SANDBOX_MODE)
+    public function __construct($apiKey, $projectId, ClientInterface $http_client)
```
Comment (dimitrystd) [inline src/SmartlingApi.php]: Please add one more constructor with `$baseUrl`

File: src/SmartlingApi.php [hunk 7]
```diff
@@ -122,19 +112,32 @@ protected function sendRequest($uri, $requestData, $method)
...
+
+            if (!$error_response || empty($error_response['response']['messages'])) {
+                throw new SmartlingApiException('Bad response format from Smartling');
+            }
+
+            throw new SmartlingApiException(implode(' || ', $error_response['response']['messages']), $guzzle_response->getStatusCode());
+        }
+
+        // "Download file" method return translated file directly.
+        if ('file/get' == $uri) {
```
Comment (gsoul) [inline src/SmartlingApi.php]: if is a good practice to use "===" instead of "==", unless there is a really good reason not to do it.

**Pull request merged**